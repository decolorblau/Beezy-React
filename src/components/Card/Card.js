import "./Card.scss";
import PropTypes from "prop-types";

const Card = ({ marvelComic }) => {
  return (
    <section className="card">
      <img
        className="card__image"
        src={`${marvelComic.thumbnail.path}.${marvelComic.thumbnail.extension}`}
        alt={`Poster of ${marvelComic.title}`}
      />
      <div className="card__info">
        <h3 className="card__title">{marvelComic.title}</h3>
        <div className="card__text">
          <p className="card__format">{marvelComic.format}</p>
          <p className="card__price">Price: {marvelComic.prices[0].price} â‚¬</p>
        </div>
      </div>
    </section>
  );
};

Card.propTypes = {
  marvelEvent: PropTypes.object,
};

export default Card;
