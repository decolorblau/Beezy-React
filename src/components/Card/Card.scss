* {
  margin: 0;
  padding: 0;
}

a {
  text-decoration: none;
}
.card {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  position: relative;
  overflow: hidden;
  margin: 5px;
  margin-bottom: 10px;
  min-height: 239px;
  &__image {
    width: 100%;
    height: auto;
    border-radius: 5px;
  }

  &__info {
    width: 100%;
    bottom: 0px;
    height: 35px;
    text-align: right;
  }

  &__title {
    font-size: 13px;
    font-weight: 500;
    color: white;
    margin-top: 5px;
    font-family: "marvelregular", sans-serif;
    letter-spacing: 1px;
  }
  &__text {
    display: none;
  }
}

.card:hover .card__title {
  color: rgb(209, 0, 0);
}

@media (min-width: 768px) {
  .card__info {
    height: 40px;
  }
  .card__title {
    font-size: 15px;
    font-weight: 500;
    color: white;
  }
}

@media (min-width: 1024px) {
  .card {
    display: flex;
    justify-content: flex-end;
    position: relative;
    overflow: hidden;
    margin: 5px;
    overflow-y: hidden;
    padding-bottom: 0px;
    &__info {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      width: 100%;
      bottom: -220px;
      font-family: "Roboto", sans-serif;
      backdrop-filter: blur(5px);
      background: rgba(21, 24, 27, 0.8);
      transition: 0.5s;
      position: absolute;
    }
    &__title {
      display: none;
    }
    &__info > div {
      margin-top: 0;
    }
    &__text {
      display: none;
    }
  }
  .card:hover .card__info {
    bottom: 0px;
    height: 100%;
    .card__title {
      display: block;
      font-size: 16px;
      color: rgb(209, 0, 0);
      font-weight: 700;
      margin-bottom: 10px;
      padding: 0 5px;
      text-align: center;
    }
    .card__text {
      display: flex;
      justify-content: center;
      align-items: center;
      color: white;
      font-size: 12px;
    }
    .card__price {
      position: absolute;
      text-align: center;
      bottom: 30px;
      color: white;
      font-family: "roboto", sans-serif;
      font-size: 16px;
    }
  }
}
