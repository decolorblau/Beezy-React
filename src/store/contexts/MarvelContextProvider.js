import MarvelContext from "./MarvelContext";

const MarvelContextProvider = ({ children }) => {
  return <MarvelContext.Provider>{children}</MarvelContext.Provider>;
};

export default MarvelContextProvider;
