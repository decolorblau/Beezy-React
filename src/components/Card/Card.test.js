import ReactTestRenderer from "react-test-renderer";
import MarvelContextProvider from "../../store/contexts/MarvelContextProvider";
import Card from "./Card";

describe("Given the component Card", () => {
  describe("When it receives a object", () => {
    test("Then it should render a card with the comic info inside", () => {
      const comic = {
        id: 2345,
        title: "Avengers",
        format: "comic",
        thumbnail: {
          path: "https://image.tmdb.org/t/p/w500",
          extension: "jpg",
        },
        prices: [{ price: 3.99 }],
      };

      const cardComponent = ReactTestRenderer.create(
        <MarvelContextProvider>
          <Card marvelComic={comic} />
        </MarvelContextProvider>
      );

      expect(cardComponent.toJSON).toMatchSnapshot();
    });
  });
});
