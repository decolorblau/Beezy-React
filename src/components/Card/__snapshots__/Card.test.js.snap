// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the component Card When it receives a object Then it should render a card with the comic info inside 1`] = `[Function]`;
